wolfcrypt/src/rsa.c

```diff
@@ -2865,23 +2867,36 @@ int cc310_RsaSSL_Verify(const byte* in, word32 inLen, byte* sig,
 int wc_RsaFunction(const byte* in, word32 inLen, byte* out,
                           word32* outLen, int type, RsaKey* key, WC_RNG* rng)
 {
+#ifndef WOLF_CRYPTO_CB_ONLY_RSA
     int ret = 0;
-
+#endif
     if (key == NULL || in == NULL || inLen == 0 || out == NULL ||
             outLen == NULL || *outLen == 0 || type == RSA_TYPE_UNKNOWN) {
         return BAD_FUNC_ARG;
     }
 
 #ifdef WOLF_CRYPTO_CB
     if (key->devId != INVALID_DEVID) {
+    #ifndef WOLF_CRYPTO_CB_ONLY_RSA
         ret = wc_CryptoCb_Rsa(in, inLen, out, outLen, type, key, rng);
         if (ret != CRYPTOCB_UNAVAILABLE)
             return ret;
         /* fall-through when unavailable */
         ret = 0; /* reset error code and try using software */
+    #else
+        /* CBONLY_TEST_DEVID is only valid for testing */
+        if(key->devId == CBONLY_TEST_DEVID) return 0;
```

Comment:
Space after `if (`. Can we find a way to remove `CBONLY_TEST_DEVID` all together? I'd prefer not having code in the library for special test case handling. Instead it should be handled in the wolfCrypt test.

Reply:
remove CBONLY_TEST_DEVID